source/common/tcp_proxy/upstream.h

```diff
@@ -77,7 +77,8 @@ class HttpConnPool : public GenericConnPool, public Http::ConnectionPool::Callba
     virtual void onFailure() {
       ASSERT(conn_pool_ != nullptr);
       conn_pool_->callbacks_->onGenericPoolFailure(
-          ConnectionPool::PoolFailureReason::RemoteConnectionFailure, host_);
+          ConnectionPool::PoolFailureReason::RemoteConnectionFailure, "remote_connection_failure",
```

Comment:
looking at the call site here, it looks like we call this when we fail a CONNECT request, which isn't really a transport failure?
I'm not sure if you consider that a transport failure (connect handshake like the TLS handshake in that case) but I think either the underlying call sites should return details through here, or if you don't consider it a transport failure we shouldn't set a string here.

Reply:
I am leaning towards not adding a new failure message here as the intent of the PR is to propagate the failure reason IF ANY.